include/moq/detail/uintvar.h

```diff
@@ -39,82 +79,66 @@ namespace qtransport {
      *
      * @returns vector of encoded bytes or empty vector if value is invalid
      */
-    inline UintVT ToUintV(uint64_t value)
+    inline UintV ToUintV(uint64_t value)
     {
-        static constexpr uint64_t kLen1 = (static_cast<uint64_t>(-1) << (64 - 6) >> (64 - 6));
-        static constexpr uint64_t kLen2 = (static_cast<uint64_t>(-1) << (64 - 14) >> (64 - 14));
-        static constexpr uint64_t kLen4 = (static_cast<uint64_t>(-1) << (64 - 30) >> (64 - 30));
-
-        uint8_t net_bytes[8]{ 0 }; // Network order bytes
-        uint8_t len{ 0 };          // Length of bytes encoded
+        constexpr uint64_t kLen1 = (static_cast<uint64_t>(-1) << (64 - 6) >> (64 - 6));
+        constexpr uint64_t kLen2 = (static_cast<uint64_t>(-1) << (64 - 14) >> (64 - 14));
+        constexpr uint64_t kLen4 = (static_cast<uint64_t>(-1) << (64 - 30) >> (64 - 30));
 
-        uint8_t* byte_value = reinterpret_cast<uint8_t*>(&value);
-
-#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
-        constexpr std::array<uint8_t, sizeof(uint64_t)> host_order{ 0, 1, 2, 3, 4, 5, 6, 7 };
-#else
-        constexpr std::array<uint8_t, sizeof(uint64_t)> kHostOrder{ 7, 6, 5, 4, 3, 2, 1, 0 };
-#endif
-
-        if (byte_value[kHostOrder[0]] & 0xC0) { // Check if invalid
+        if (value & (0xC0ull << 56)) { // Check if invalid
             return {};
         }
 
+        std::size_t length = sizeof(uint8_t);
         if (value > kLen4) { // 62 bit encoding (8 bytes)
-            for (int i = 0; i < 8; i++) {
-                net_bytes[i] = byte_value[kHostOrder[i]];
-            }
-            net_bytes[0] |= 0xC0;
-            len = 8;
+            value |= (0xC0ull << 56);
+            length = sizeof(uint64_t);
         } else if (value > kLen2) { // 30 bit encoding (4 bytes)
-            for (int i = 0; i < 4; i++) {
-                net_bytes[i] = byte_value[kHostOrder[i + 4]];
-            }
-            net_bytes[0] |= 0x80;
-            len = 4;
+            value |= (0x80u << 24);
+            length = sizeof(uint32_t);
         } else if (value > kLen1) { // 14 bit encoding (2 bytes)
-            net_bytes[0] = byte_value[kHostOrder[6]] | 0x40;
-            net_bytes[1] = byte_value[kHostOrder[7]];
-            len = 2;
-        } else {
-            net_bytes[0] = byte_value[kHostOrder[7]];
-            len = 1;
+            value |= (0x40u << 8);
+            length = sizeof(uint16_t);
         }
 
-        std::vector<uint8_t> encoded_bytes(net_bytes, net_bytes + len);
-        return encoded_bytes;
+        uint64_t net_value = SwapBytes(value);
```

Comment:
Not sure how much more we can optimize the vector, since we have to create one for uintvar_t to work at all. We only create one as long as we need, and memcpy into it (about as fast as we possibly can). The operation only takes about 20-25ns, not sure we can ask any more of a vector. We can though avoid storing another local variable though, so we can just set `value = SwapBytes(value)`, which should be ever so slightly cheaper.

Reply:
Maybe yours is isn’t g release build. Mine is >200ns just for vector construct.